Merge hyperboria/next into crashey

lgierth opened the pull request:
First PR from https://github.com/hyperboria/cjdns

- two seccomp fixes by @meshpoint
- updates from docs.git

cjdelisle commented on doc/cjdns/upcoming/iface-h.md:
```diff
@@ -0,0 +1,10 @@
+
+If anybody wants to get a leg up on an upcoming feature:
+
+check out the new [Iface.h](https://github.com/cjdelisle/cjdns/blob/ducttape2/interface/Iface.h) which will replace the current `Interface.h`.
```
already in crashey and I think master...

cjdelisle commented on doc/faq/security.md:
```diff
@@ -0,0 +1,30 @@
+
+## If you're hosting a cjdns node on a VPS
+
+There are quite likely measures that you can take to encrypt your VPS's hard drive, and you should look into that, but suppose you haven't encrypted anything. In that case, yes, your VPS provider can probably just go and look at your private keys, and read your encrypted traffic.
```
this is misleading, if you have a VPS, your provider can pause you for a moment and take a snapshot of your ram, then dump your keys. There's no way around this, people write papers about how to protect against this, in the end you always lose. However PFS should be taken into account. If you lose your cjdroute.conf file, it doesn't mean you lost all sessions prior because temp keys are used and destroyed. It does mean someone can impersonate you later but it also doesn't mean they can just begin decrypting your traffic at will. Keys are rotated when CA sessions timeout which is "random" but obviously they are all killed when cjdroute is restarted.

cjdelisle commented on util/Seccomp.c:
```diff
@@ -38,7 +38,7 @@
 {
     printf("Attempted banned syscall number [%d] see doc/Seccomp.md for more information\n",
-           si->si_value.sival_int);
+           si->si_syscall);
```
Good idea, si_syscall was added in 3.5 and so was SECCOMP_BPF so it's safe to use in this file.

cjdelisle commented on doc/LEGAL.md:
```diff
@@ -0,0 +1,7 @@
+Documentation (c) by ProjectMeshnet.
```
Is this a legal entity? Do you have CLA signatures from the actual contributors of the documentation? If no to either then this statement is technically wrong. However you can state that all documentation is CC-BY-4.0 and it should be fair to assume that anyone contributing documentation has agreed to license /their/ copyright under this license, as I do with the GPLv3+ and the codebase.

lgierth commented:
I think we addressed your comments, I just performed another subtree merge of docs.git. The squash commits from `git subtree merge --squash` are really neat -- they contain a `git log --oneline` for the squashed commit range.
